File: app/src/main/java/dev/ragnarok/fenrir/model/Token.java
package dev.ragnarok.fenrir.model;

import android.os.Parcel;
import android.os.Parcelable;


public final class Token implements Parcelable {

    public static final Creator<Token> CREATOR = new Creator<Token>() {
        @Override
        public Token createFromParcel(Parcel in) {
            return new Token(in);
        }

        @Override
        public Token[] newArray(int size) {
            return new Token[size];
        }
    };
    private final int ownerId;
    private final String accessToken;

    public Token(int ownerId, String accessToken) {
        this.ownerId = ownerId;
        this.accessToken = accessToken;
    }

    private Token(Parcel in) {
        ownerId = in.readInt();
        accessToken = in.readString();
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeInt(ownerId);
        dest.writeString(accessToken);
    }

    public int getOwnerId() {
        return ownerId;
    }

    public String getAccessToken() {
        return accessToken;
    }

    @Override
    public String toString() {
        return "Token{" +
                "ownerId=" + ownerId +
                ", accessToken='" + accessToken + '\'' +
                '}';
    }
}

File: app_fenrir/src/main/kotlin/dev/ragnarok/fenrir/model/Token.kt
package dev.ragnarok.fenrir.model

import android.os.Parcel
import android.os.Parcelable

class Token : Parcelable {
    val ownerId: Long
    val accessToken: String?

    constructor(ownerId: Long, accessToken: String?) {
        this.ownerId = ownerId
        this.accessToken = accessToken
    }

    internal constructor(parcel: Parcel) {
        ownerId = parcel.readLong()
        accessToken = parcel.readString()
    }

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeLong(ownerId)
        dest.writeString(accessToken)
    }

    override fun toString(): String {
        return "Token{" +
                "ownerId=" + ownerId +
                ", accessToken='" + accessToken + '\'' +
                '}'
    }

    companion object CREATOR : Parcelable.Creator<Token> {
        override fun createFromParcel(parcel: Parcel): Token {
            return Token(parcel)
        }

        override fun newArray(size: Int): Array<Token?> {
            return arrayOfNulls(size)
        }
    }
}